rs/p2p/artifact_downloader/src/fetch_stripped_artifact/assembler.rs

```diff
@@ -222,33 +227,222 @@ async fn get_or_fetch<P: Peers>(
     .await
```

Comment:
Good question. 

Currently, we will keep trying to download the ingress message (with a backoff) but will keep failing, so we will never fully reassemble the block.

There are only two scenarios when this could happen:
1) The block we are trying to reassemble is not advertised anymore by anyone - in which case our `assembly` task will be cancelled by the caller [here](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/p2p/consensus_manager/src/receiver.rs?L510-515)
2) A malicious block maker proposed a block with non-existing ingress message - in which case no node in the subnet will be able to reconstruct & validate the block, meaning some other block should be eventually notarized and finalized at the same height, at which point the priority function will tell us to `Drop` the block we are trying to reconstruct, similarly to how it's done [here](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/p2p/artifact_downloader/src/fetch_artifact/download.rs?L163-165). This hasn't been implemented yet, but will be in a subsequent PR ([jira ticket here](https://dfinity.atlassian.net/browse/CON-1401))

Reply:
Great, thanks